Library/Homebrew/cmd/vendor-install.sh

```diff
@@ -14,36 +14,37 @@ VENDOR_DIR="${HOMEBREW_LIBRARY}/Homebrew/vendor"
 
 # Built from https://github.com/Homebrew/homebrew-portable-ruby.
 set_ruby_variables() {
-  if [[ -n "${HOMEBREW_MACOS}" ]]
+  # Handle the case where /usr/local/bin/brew is run under arm64.
+  # It's a x86_64 installation there (we refuse to install arm64 binaries) so
+  # use a x86_64 Portable Ruby.
+  if [[ -n "${HOMEBREW_MACOS}" && "${VENDOR_PHYSICAL_PROCESSOR}" == "arm64" && "${HOMEBREW_PREFIX}" == "/usr/local" ]]
   then
-    if [[ "${VENDOR_PHYSICAL_PROCESSOR}" == "x86_64" ]] ||
-       # Handle the case where /usr/local/bin/brew is run under arm64.
-       # It's a x86_64 installation there (we refuse to install arm64 binaries) so
-       # use a x86_64 Portable Ruby.
-       [[ "${VENDOR_PHYSICAL_PROCESSOR}" == "arm64" && "${HOMEBREW_PREFIX}" == "/usr/local" ]]
+    ruby_PROCESSOR="x86_64"
+    ruby_OS="darwin"
+  else
+    ruby_PROCESSOR="${VENDOR_PHYSICAL_PROCESSOR}"
```

Comment:
If `VENDOR_PHYSICAL_PROCESSOR` is used for a file input here: may want to ensure it's only ever set by this script and otherwise e.g. unset to avoid it being set to arbitrary values?

Reply:
Perfect, thanks!